drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/HeaderTransport.java

```diff
@@ -184,10 +227,10 @@ public static ThriftFrame decodeFrame(ByteBuf buffer)
             allHeaders.putAll(decodeHeaders(PERSISTENT_HEADERS, messageHeader));
 
             // message
-            ByteBuf message = buffer.readBytes(buffer.readableBytes());
+            ByteBuf message = untransform(buffer.retainedDuplicate(), readTransforms);
```

Comment:
curious: What's `retainedDuplicate ` ?

Reply:
ByteBuf is a wrapper around some memory region, that has 2 pointers: writePointer and readPointer. Also it keeps the reference counting. `retainedDuplicate` creates a duplicate of the wrapper and increments the memory region reference count by 1.